discordoauth2/__init__.py

```diff
@@ -1,296 +1,12 @@
-        """Revokes a OAuth2 token related to the client.
-        
-        token: the token string to be revoked.
-        token_type: the type of token to be revoked. This may be ignored by Discord.
-        """
-        response = requests.post("https://discord.com/api/oauth2/token/revoke",
-            data={"token": token, "token_type_hint": token_type},
-            auth=(self.id, self.__secret))
-        print(response.status_code, response.text)
-        if response.ok:
-            return
-        elif response.status_code == 401: raise exceptions.Forbidden(f"this AccessToken does not have the nessasary scope.")
-        elif response.status_code == 429: raise exceptions.RateLimited(f"You are being Rate Limited. Retry after: {response.json()['retry_after']}", retry_after=response.json()['retry_after'])
-        else:
-            raise exceptions.HTTPException(f"Unexpected HTTP {response.status_code}")
-    
-    def generate_uri(self, scope: Union[str, list[str]], state: Optional[str]=None, skip_prompt: Optional[bool]=False, response_type: Optional[Literal["code", "token"]]="code", guild_id: Optional[Union[int, str]]=None, disable_guild_select: Optional[bool]=None, permissions: Optional[Union[int, str]]=None) -> str:
-        """Creates an authorization uri with client information prefilled.
-        
-        scope: a string, or list of strings for the scope
-        state: optional state parameter. Optional but recommended.
-        skip_prompt: doesn't require the end user to reauthorize if they've already authorized you app before. Defaults to `False`.
-        response_type: either code, or token. token means the server can't access it, but the client can use it without converting.
-        guild_id: the guild ID to add a bot/webhook.
-        disable_guild_select: wether to allow the authorizing user to change the selected guild
-        permissions: the permission bitwise integer for the bot being added.
-        """
-        params = {
-            "client_id": self.id,
-            "scope": " ".join(scope) if type(scope) == list else scope,
-            "state": state,
-            "redirect_uri": self.redirect_url,
-            "prompt": "none" if skip_prompt else None,
-            "response_type": response_type,
-            "guild_id": guild_id,
-            "disable_guild_select": disable_guild_select,
-            "permissions": permissions
-        }
-        return f"https://discord.com/oauth2/authorize?{parse.urlencode({key: value for key, value in params.items() if value is not None})}"
-
-class exceptions():
-    class BaseException(Exception):
-        pass
-
-    class HTTPException(BaseException):
-        pass
-
-    class RateLimited(HTTPException):
-        def __init__(self, text, retry_after):
-            self.retry_after = retry_after
-            super().__init__(text)
-  
-    class Forbidden(HTTPException):
-        pass
+from .async_oauth import AsyncClient, AsyncAccessToken, AsyncPartialAccessToken
```

Comment:
why don't we import exceptions from the async version, wouldn't this cause issues with error handling with aysnc?

Reply:
we should separate the exceptions to a different file for ease of organization;

Reply:
I actually wanted to add that to the pr then though I should stop working for the day went to sleep then didn't sleep and saw my emails. Il will change that tomorrow.

Reply:
Cuz I saw how bad I did that part but was too tired to work on it